app/(admin)/admin.tsx

```diff
@@ -1,56 +1,251 @@
 //import liraries
-import React, { Component } from 'react';
-import { View, Text, StyleSheet, Button, Pressable } from 'react-native';
+import React, { Component, useState } from 'react';
+import { View, Text, StyleSheet, Button, Pressable, Modal, TextInput } from 'react-native';
 import { LinearGradient } from 'expo-linear-gradient';
+import GameModal from './components/GameModal';
+import { FIRESTORE } from '../../FirebaseConfig';
+import { addDoc, collection, doc, Firestore, setDoc, Timestamp, updateDoc } from 'firebase/firestore';
+import getTeam from '../../services/teamFetcher';
+import { Team } from '../../types/Team';
+import { Game } from '../../types/Game';
+import DateTimePicker, { DateTimePickerEvent } from '@react-native-community/datetimepicker'
+
 
 // create a component
 const AdminPage = () => {
-    const createGame = async () => {
+    const db = FIRESTORE;
+    // create game states
+    const [createGameModalVisible, setCreateGameModalVisible] = useState(false);
+    const [team1ID, setTeam1ID] = useState('');
+    const [team2ID, setTeam2ID] = useState('');
+    const [date, setDate] = useState(new Date());
+    // create team states
+    const [createTeamModalVisible, setCreateTeamModalVisible] = useState(false);
+    const [teamName, setTeamName] = useState('');
+    const [color1, setColor1] = useState('');
+    const [color2, setColor2] = useState('');
+    const [logo, setLogo] = useState('');
+    //  edit team states
+    const [editTeamModalVisible, setEditTeamModalVisible] = useState(false);
+    const [oldTeamID, setOldTeamID] = useState('');
+    const [newTeamName, setNewTeamName] = useState('');
+    const [newColor1, setNewColor1] = useState('');
+    const [newColor2, setNewColor2] = useState('');
+    const [newLogo, setNewLogo] = useState('');
 
-    }
+    const onChangeDate = (event: DateTimePickerEvent, selectedDate?: Date) => {
+        const currentDate = selectedDate || date;
+        setDate(currentDate);
+    };
 
-    const editGame = async () => {
-        
-    }
+    const onChangeTime = (event: DateTimePickerEvent, selectedTime?: Date) => {
+        const currentTime = selectedTime || date;
+        setDate(currentTime);
+    };
 
-    const deleteGame = async () => {
-        
+    const handleAddGame = async (db: Firestore, team1ID: String, team2ID: String, startTime: Timestamp) => {
+        if (team1ID && team2ID && startTime) {
+    
+            const docRef = await addDoc(collection(db, "games"), {
+                gameID: '',
```

Comment:
After testing, the gameID field is automatically updated. Not sure why, but it doesn't require an additional call

Reply:
UPDATE: I see now that you have something to address by adding the id after creation. I'm not able to test this on Android because of the bug with datepicker, but the new document in Firestore does not have an id. You can ignore this comment because I can't be sure it works or not